netty/src/main/java/io/grpc/netty/UdsNameResolver.java

```diff
@@ -57,8 +60,9 @@ private void resolve() {
     ResolutionResult.Builder resolutionResultBuilder = ResolutionResult.newBuilder();
     List<EquivalentAddressGroup> servers = new ArrayList<>(1);
     servers.add(new EquivalentAddressGroup(new DomainSocketAddress(authority)));
-    resolutionResultBuilder.setAddresses(Collections.unmodifiableList(servers));
-    listener.onResult(resolutionResultBuilder.build());
+    resolutionResultBuilder.setAddressesOrError(StatusOr.fromValue(servers));
+    syncContext.execute(() ->
```

Comment:
Why this change? start and refresh must be called from the synchronization context.

Reply:
Done.